query/src/main/java/org/infinispan/query/impl/EntityLoader.java

```diff
@@ -23,20 +25,29 @@ public EntityLoader(AdvancedCache<?, ?> cache, KeyTransformationHandler keyTrans
       this.cache = cache;
    }
 
+   private Object decodeKey(EntityInfo entityInfo) {
+      return keyTransformationHandler.stringToKey(entityInfo.getId().toString(), cache.getClassLoader());
+   }
+
    public Object load(EntityInfo entityInfo) {
-      Object cacheKey = keyTransformationHandler.stringToKey(entityInfo.getId().toString(), cache.getClassLoader());
-      return cache.get(cacheKey);
+      return cache.get(decodeKey(entityInfo));
    }
 
-   public List<Object> load(Collection<EntityInfo> entityInfos) {
-      ArrayList<Object> list = new ArrayList<Object>(entityInfos.size());
+   public List<Object> load(List<EntityInfo> entityInfos) {
+      int entitiesSize = entityInfos.size();
+      Set<Object> keys = new LinkedHashSet<>(entitiesSize);
```

Comment:
Ok sorry I see you actually iterate on it. My bad, I expected the iteration to happen on _entityInfos_